src/components/Popup/Popup.tsx

```diff
@@ -0,0 +1,197 @@
+  | 'end center'
+  | 'top center'
+  | 'bottom center'
+
+const POSITIONS: PopupPosition[] = [
+  'top start',
+  'top end',
+  'bottom start',
+  'bottom end',
+  'start center',
+  'end center',
+  'top center',
+  'bottom center',
+]
+
+export interface IPopupProps extends IPortalGenericProps {
+  as?: any
+  basic?: boolean
+  className?: string
+  content?: ItemShorthand | ItemShorthand[]
+  position?: PopupPosition
+  style?: CSSProperties
+  trigger: JSX.Element
+}
+
+export interface IPopupState extends IPortalGenericState {
+  style?: CSSProperties
+}
+
+/**
+ * A Popup displays additional information on top of a page.
+ * @accessibility This is example usage of the accessibility tag.
+ * This should be replaced with the actual description after the PR is merged
+ */
+export default class Popup extends PortalGeneric<Extendable<IPopupState>, IPopupState> {
+  private popupOffset = 8
+  private popupCoords: ClientRect | DOMRect | undefined
+
+  public static Content = PopupContent
+
+  public static propTypes = {
+    /** An element type to render as (string or function). */
+    as: customPropTypes.as,
+
+    /** Basic CSS styling for the popup */
+    basic: PropTypes.bool,
+
+    /** Primary content. */
+    children: PropTypes.node,
+
+    /** Additional classes. */
+    className: PropTypes.string,
+
+    /** Simple text content for the popover. */
+    content: customPropTypes.itemShorthand,
+
+    /** Position for the popup. */
+    position: PropTypes.oneOf(POSITIONS),
+
+    /** Element to be rendered in-place where the popup is defined. */
+    trigger: PropTypes.node.isRequired,
+
+    /** Custom style to be applied for component. */
+    style: PropTypes.object,
+
+    /**
+     * Called when the portal is mounted on the DOM.
+     *
+     * @param {object} data - All props.
+     */
+    onMount: PropTypes.func,
+
+    /**
+     * Called when the portal is unmounted from the DOM.
+     *
+     * @param {object} data - All props.
+     */
+    onUnmount: PropTypes.func,
+
+    /** Controls whether or not the portal is displayed. */
+    open: PropTypes.bool,
+
+    /** Initial open value */
+    defaultOpen: PropTypes.bool,
+  }
+
+  public static defaultProps = {
+    position: 'top start',
+  }
+
+  public state = { style: {}, open: false }
+
+  public renderComponent({  }: IRenderResultConfig<IPopupProps>): React.ReactNode {
+    const { basic, children, content, trigger } = this.props
+    const { style } = this.state
+
+    const portalContent = (
+      <Popup.Content basic={basic} triggerRef={this.handlePopupRef} styles={{ root: style }}>
+        {childrenExist(children) ? children : content}
+      </Popup.Content>
+    )
+
+    return (
+      <React.Fragment>
+        {this.renderPortal(portalContent)}
+        {this.renderTrigger(trigger)}
+      </React.Fragment>
+    )
+  }
+
+  protected handleMount() {
+    super.handleMount()
+    this.setPopupStyle()
+  }
+
+  private computePopupStyle(): CSSProperties {
```

Comment:
this will be removed after the approach `react-popper` will become stable

Reply:
yep, remember that